src/Microsoft.Extensions.Configuration.AzureAppConfiguration/IConfigurationClient.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+
+using Azure;
+using Azure.Data.AppConfiguration;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
+{
+    internal interface IConfigurationClient
```

Comment:
This naming is so confusing. I would have guessed this is an interface defined in the SDK instead of the provider.

I don't see the need of this interface. Everything is internal. We could just use the concrete type in the AACOptions and AACProvider classes.